tests/unit/test_rest_streaming.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import random
+from typing import List
+from unittest.mock import patch
+
+import proto
+import pytest
+import requests
+
+from google.api_core import rest_streaming
+
+
+class Composer(proto.Message):
+    given_name = proto.Field(proto.STRING, number=1)
+    family_name = proto.Field(proto.STRING, number=2)
+
+
+class Song(proto.Message):
+    composer = proto.Field(Composer, number=1)
+    title = proto.Field(proto.STRING, number=2)
+    lyrics = proto.Field(proto.STRING, number=3)
+    year = proto.Field(proto.INT32, number=4)
+
+
+class EchoResponse(proto.Message):
+    content = proto.Field(proto.STRING, number=1)
+
+
+class ResponseMock(requests.Response):
+    class _ResponseItr:
+        def __init__(
+            self,
+            _response_bytes: bytes,
+            decode_unicode: bool,
+            random_split=False,
+            seed=0,
+        ):
+            self._responses_bytes = _response_bytes
+            self._i = 0
+            self._decode_unicode = decode_unicode
+            self._random_split = random_split
+            random.seed(seed)
+
+        def __next__(self):
+            if self._i == len(self._responses_bytes):
+                raise StopIteration
+            if self._random_split:
+                n = random.randint(1, len(self._responses_bytes[self._i :]))
+            else:
+                n = 1
+            x = self._responses_bytes[self._i : self._i + n]
+            self._i += n
+            if self._decode_unicode:
+                x = x.decode("utf-8")
+            return x
+
+        def __iter__(self):
+            return self
+
+    def __init__(
+        self,
+        *args,
+        responses: List[proto.Message],
+        random_split=False,
+        response_cls,
+        **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        self._responses = responses
+        self._random_split = random_split
+        self._response_message_cls = response_cls
+
+    def _parse_responses(self, responses: List[proto.Message]) -> bytes:
+        ret_val = "["
+        # json.dumps returns a string surrounded with quotes that need to be stripped
+        # in order to be an actual JSON.
+        json_responses = [
+            json.dumps(self._response_message_cls.to_json(r))[1:-1] for r in responses
```

Comment:
What do you think about
```python
json.dumps(self._response_message_cls.to_json(r)).strip('"') for r in responses
```
?

I'm on the fence, keep it the way it is if you think it's clearer.

Reply:
Done